h2o-automl/src/main/java/ai/h2o/automl/AutoML.java

```diff
@@ -684,671 +493,39 @@ private int nextInstanceCounter(String algoName, NonBlockingHashMap<String, Inte
-    double maxAssignedTime = timeRemainingMs() * workAllocations.remainingWorkRatio(work) / 1e3;
-    // predicate can be removed if/when we decide to include SEs in the max_models limit
-    int maxAssignedModels = (int) Math.ceil(remainingModels() * workAllocations.remainingWorkRatio(work, new Predicate<WorkAllocations.Work>() {
-      @Override
-      public Boolean apply(WorkAllocations.Work work) {
-        return work.algo != Algo.StackedEnsemble;
-      }
-    }));
-
-    if (searchCriteria.max_runtime_secs() == 0)
-      searchCriteria.set_max_runtime_secs(maxAssignedTime);
-    else
-      searchCriteria.set_max_runtime_secs(Math.min(searchCriteria.max_runtime_secs(), maxAssignedTime));
-
-    if (searchCriteria.max_models() == 0)
-      searchCriteria.set_max_models(maxAssignedModels);
-    else
-      searchCriteria.set_max_models(Math.min(searchCriteria.max_models(), maxAssignedModels));
-
-    eventLog().info(Stage.ModelTraining, "AutoML: starting "+algo+" hyperparameter search");
-
-    if (null == gridKey) gridKey = gridKey(algo.name());
-    addGridKey(gridKey);
-    Log.debug("Hyperparameter search: "+algo.name()+", time remaining (ms): "+timeRemainingMs());
-    return GridSearch.startGridSearch(
-        gridKey,
-        baseParms,
-        searchParms,
-        new GridSearch.SimpleParametersBuilderFactory<>(),
-        searchCriteria
-    );
-  }
-
-  Job<StackedEnsembleModel> stack(String modelName, Key<Model>[] modelKeyArrays, boolean use_cache) {
-    WorkAllocations.Work work = workAllocations.getAllocation(Algo.StackedEnsemble, JobType.ModelBuild);
-    if (work == null) return null;
-
-    // Set up Stacked Ensemble
-    StackedEnsembleParameters stackedEnsembleParameters = new StackedEnsembleParameters();
-    stackedEnsembleParameters._base_models = modelKeyArrays;
-    stackedEnsembleParameters._valid = (getValidationFrame() == null ? null : getValidationFrame()._key);
-    stackedEnsembleParameters._blending = (getBlendingFrame() == null ? null : getBlendingFrame()._key);
-    stackedEnsembleParameters._keep_levelone_frame = true; //TODO Why is this true? Can be optionally turned off
-    stackedEnsembleParameters._keep_base_model_predictions = use_cache; //avoids recomputing some base predictions for each SE
-    // Add cross-validation args
-    stackedEnsembleParameters._metalearner_fold_column = buildSpec.input_spec.fold_column;
-    stackedEnsembleParameters._metalearner_nfolds = buildSpec.build_control.nfolds;
-
-    stackedEnsembleParameters.initMetalearnerParams();
-    stackedEnsembleParameters._metalearner_parameters._keep_cross_validation_models = buildSpec.build_control.keep_cross_validation_models;
-    stackedEnsembleParameters._metalearner_parameters._keep_cross_validation_predictions = buildSpec.build_control.keep_cross_validation_predictions;
-
-    Key modelKey = modelKey(modelName, false);
-    return trainModel(modelKey, work, stackedEnsembleParameters, true);
-  }
-
```

Comment:
`setCommonModelBuilderParams` and `setStoppingCriteria` logic moved with barely no changes to abstract class `TrainingStep`